guides/hack/05-functions/30-function-references.md

```diff
@@ -0,0 +1,183 @@
+It is often useful to refer to a function
+(or a [method](https://docs.hhvm.com/hack/classes/methods))
+without actually calling it&mdash;for example,
+as an argument for functions like `Vec\map`.
+
+**Note:** The following syntax is only supported since HHVM 4.79. For older HHVM
+version, see [old syntax](#old-syntax) below.
+
+To refer to a top-level (global) function named `foo`, you can write:
+
+```
+foo<>; // a reference to global function 'foo'
+```
+
+You can think of it like it’s a function call with empty generics, but the list
+of arguments has been omitted (the missing parentheses).
+
+The following example stores a function reference in a variable and later calls
+the function. Note that the type checker keeps track of the
+[function's type](/hack/built-in-types/function)
+and correctly checks all calls.
+
+```
+function foo(string $x): bool { ... }
+
+$x = foo<>;     // $x:(function(string): bool)
+$y = $x('bar'); // $y:bool
+$_ = $x(-1);    // error!
+```
+
+This syntax supports namespaced names the same way you would refer to them as
+part of a function call, so the following function references are all
+equivalent:
+
+```
+$fun = \Foo\Bar\Baz\derp<>;
+```
+
+```
+namespace Foo;
+$fun = Bar\Baz\derp<>;
+```
+
+```
+use namespace Foo\Bar\Baz;
+$fun = Baz\derp<>;
+```
+
+```
+use function Foo\Bar\Baz\derp;
+$fun = derp<>;
+```
+
+## Static methods
+
+Similarly, you can refer to a static method `bar` on a class `MyClass` by using
+the familiar method call syntax, without providing the call arguments. Just
+append type arguments (like `<>`) to the function call receiver
+(like `MyClass::bar`).
+
+```
+MyClass::bar<>; // a reference to static method 'MyClass::bar'
+```
+
+- **Private/protected** methods can be referenced using this syntax as long as
+  they are accessible to you in your local scope.
```

Comment:
Once you have such a reference to a private/protected method, who can invoke it?